### PR TITLE
[Ask Mode] re-arranged parameter to keep binary back compat

### LINKED ISSUE #9047: CompilationWithAnalyzerOptions ctor must remove breaking change
Consider this commit:  https://github.com/dotnet/roslyn/commit/deeb46ecd3f215564f344f0f42bba90adff255e5

It changed the parameters on `CompilationWithAnalyzerOptions`.  This is a breaking change from update 1.  It needs to be adjusted to maintain the old constructors in addition to the new ones to maintain binary compat.

### PR DESCRIPTION
**Pull request by heejaechang**
fix #9047

...

this is to keep binary backward compatibility. new parameter gives a way for third party analyzer author to crash host when exception happened so that they can get a better repro/dump for their analyzer.

in VS, we control the value for the parameter, so third-party analyzer can't crash VS.

### CODE REVIEW

**Comment by heejaechang** [comment]
@davkean @jaredpar @mavasani  ported to stabilization branch

**Comment by heejaechang** [comment]
opened a bug for failed test - https://github.com/dotnet/roslyn/issues/9367

**Comment by heejaechang** [comment]
retest prtest/win/dbg/unit64 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit64/4637/
// Retest reason: failure is not related to mine. opened a bug.

**Comment by heejaechang** [comment]
@MattGertz can I get approval for update 2?

**Comment by heejaechang** [comment]
@davkean @jaredpar @srivatsn @mavasani can I get sign off?

File: src/Compilers/Core/Portable/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -9,8 +9,9 @@ Microsoft.CodeAnalysis.Diagnostics.CompilationStartAnalysisContext.RegisterOpera
...
-Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzersOptions.CompilationWithAnalyzersOptions(Microsoft.CodeAnalysis.Diagnostics.AnalyzerOptions options, System.Action<System.Exception, Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer, Microsoft.CodeAnalysis.Diagnostic> onAnalyzerException, System.Func<System.Exception, bool> analyzerExceptionFilter, bool concurrentAnalysis, bool logAnalyzerExecutionTime, bool reportSuppressedDiagnostics) -> void
+Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzersOptions.CompilationWithAnalyzersOptions(Microsoft.CodeAnalysis.Diagnostics.AnalyzerOptions options, System.Action<System.Exception, Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer, Microsoft.CodeAnalysis.Diagnostic> onAnalyzerException, bool concurrentAnalysis, bool logAnalyzerExecutionTime) -> void
```
Comment (mavasani) [inline src/Compilers/Core/Portable/PublicAPI.Unshipped.txt]: I thought the first 2 APIs here were already shipped in Update1, isn't that so?

**Comment by mavasani** [comment]
:+1:

File: src/Compilers/Core/Portable/PublicAPI.Unshipped.txt (hunk 1 above)
Comment (heejaechang) [inline src/Compilers/Core/Portable/PublicAPI.Unshipped.txt]: I dont know but it is not in the shipped.txt

**Pull request merged**